py/selenium/webdriver/support/page_objects/find_all_by.py

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from selenium.common.exceptions import InvalidSelectorException, WebDriverException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.page_objects.page_object import PageObject
+from selenium.webdriver.support.wait import WebDriverWait
+
+class FindAllBy(object):
+    def __init__(self, by=By.ID, value=None):
+        """
+        Creates a new instance of FindAllBy. FindAllBy is used explicitly on PageObjects.
+        Takes a locator value and upon __get__ returns any matching elements from the DOM
+        Will wait 100 seconds for it to be visible and raise a TimeoutException if not found.
+
+        :Args:
+         - driver : A WebDriver instance
+         - by : A locator type (By.ID, By.CLASS_NAME)
+         - value : A locator value
+
+        Example:
+
+        .. code-block:: python
+
+            from selenium.webdriver.common.by import By
+            from selenium.webdriver.support.page_objects import PageObject, FindBy, FindAllBy
+
+            class Home(PageObject):
+                # Must be used on a PageObject
+                elements = FindAllBy(By.ID, 'qs')
+                more_elements = FindAllBy(By.CLASS_NAME, 'elementsList')
+
+        throws TimeoutException if nothing can be found via the specified locator values.
+
+        :rtype: list(WebElement)
+        """
+        if not By.is_valid(by) or not isinstance(value, str):
+            raise InvalidSelectorException("Invalid locator values passed in")
+
+        self.by = by
+        self.value = value
+
+    def __get__(self, obj, *args):
+        if not isinstance(obj, PageObject):
+            raise WebDriverException("FindAllBy can only be used on a PageObjects")
+        self.elements = WebDriverWait(obj.driver, 100).until(
```

Comment:
I think a timeout exception is fine, so that the end user can write their own handling. Otherwise, we are making more decisions on behalf of the end user, which means you will have to put more custom code to change behavior.